sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -702,7 +702,12 @@ def _fit_stochastic(
     def _update_no_improvement_count(self, early_stopping, X_val, y_val):
         if early_stopping:
             # compute validation score, use that for stopping
-            self.validation_scores_.append(self._score(X_val, y_val))
+            try:
+                val_score = self._score(X_val, y_val)
+            except ValueError:
```

Comment:
Since we could catch many `ValueError`, here I assume that we have something in the exception that tell us that this is a `np.nan` or something like this, isn't it?

Reply:
good idea, i'll match on that